sample/src/main/kotlin/com/slack/circuit/sample/home/CompositeScreen.kt

```diff
@@ -0,0 +1,194 @@
+    if (screen is CompositeScreen) return compositePresenterFactory.create(navigator)
+    return null
+  }
+}
+
+class CompositePresenter
+@AssistedInject
+constructor(
+  @Assisted private val navigator: Navigator,
+  private val homePresenter: HomePresenter,
+  petListPresenterFactory: PetListPresenter.Factory
+) : Presenter<CompositeScreen.CompositeState, CompositeScreen.CompositeEvent> {
+  private val petListPresenter = petListPresenterFactory.create(navigator)
+
+  @SuppressLint("FlowOperatorInvokedInComposition")
+  @Composable
+  override fun present(
+    events: Flow<CompositeScreen.CompositeEvent>
+  ): CompositeScreen.CompositeState {
+    val homeState =
+      homePresenter.present(
+        events.filterIsInstance<CompositeScreen.CompositeEvent.CompositeHomeEvent>().map {
+          it.event
+        }
+      )
+    val petListState =
+      petListPresenter.present(
+        events.filterIsInstance<CompositeScreen.CompositeEvent.CompositePetListEvent>().map {
+          it.event
+        }
+      )
+
+    return CompositeScreen.CompositeState(homeState, petListState)
+  }
+
+  @AssistedFactory
+  interface Factory {
+    fun create(navigator: Navigator): CompositePresenter
+  }
+}
+
+@ContributesMultibinding(AppScope::class)
+class CompositeScreenFactory @Inject constructor() : ScreenViewFactory {
+  override fun createView(screen: Screen): ScreenView? {
+    if (screen is CompositeScreen) {
+      return ScreenView(compositeUi())
+    }
+    return null
+  }
+}
+
+private fun compositeUi() =
+  ui<CompositeScreen.CompositeState, CompositeScreen.CompositeEvent> { state, events ->
+    CompositeScreen(state, events)
+  }
+
+@Composable
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+fun CompositeScreen(
+  state: CompositeScreen.CompositeState,
+  eventSink: (CompositeScreen.CompositeEvent) -> Unit
+) {
+  Scaffold(
+    modifier = Modifier.systemBarsPadding().fillMaxWidth(),
+    topBar = {
+      CenterAlignedTopAppBar(
+        title = {
+          Text("Adoptables", fontSize = 22.sp, color = MaterialTheme.colorScheme.onPrimaryContainer)
+        },
+        colors =
+          TopAppBarDefaults.centerAlignedTopAppBarColors(
+            containerColor = MaterialTheme.colorScheme.primaryContainer
+          )
+      )
+    },
+    bottomBar = {
+      BottomNavigationBar(selectedIndex = state.homeState.index) { index ->
+        eventSink(
+          CompositeScreen.CompositeEvent.CompositeHomeEvent(HomeScreen.Event.NavClickEvent(index))
+        )
+      }
+    },
+    content = {
+      if (state.homeState.index == DOGS_SCREEN_INDEX) {
+        PetList(state.petListState) { event ->
+          eventSink(CompositeScreen.CompositeEvent.CompositePetListEvent(event))
+        }
+      } else if (state.homeState.index == ABOUT_SCREEN_INDEX) {
+        About()
+      }
+    }
+  )
+}
+
+@Composable
+fun BottomNavigationBar(selectedIndex: Int, onSelectedIndex: (Int) -> Unit) {
```

Comment:
nit: let's make these private or internal by default for good visibility hygiene